core/storage/btree.rs

```diff
@@ -4228,6 +4231,113 @@ impl BTreeCursor {
+             ** this page contains countable entries. Increment the entry counter
+             ** accordingly.
+             */
+            // TODO: (pedrocarlo) does not take into account if "tree is not an int-key tree" condition
+            if contents.is_leaf() {
+                self.count += contents.cell_count();
+            }
+
+            let cell_idx = self.stack.current_cell_index() as usize;
+
+            // Second condition is necessary in case we return if the page is locked in the loop below
+            if contents.is_leaf() || cell_idx > contents.cell_count() {
+                loop {
+                    if !self.stack.has_parent() {
+                        // All pages of the b-tree have been visited. Return successfully
+                        self.move_to_root();
+
+                        return Ok(CursorResult::Ok(self.count));
+                    }
+
+                    // Move to parent
+                    self.going_upwards = true;
+                    self.stack.pop();
+
+                    mem_page_rc = self.stack.top();
+                    return_if_locked_maybe_load!(self.pager, mem_page_rc);
+                    mem_page = mem_page_rc.get();
+                    contents = mem_page.contents.as_ref().unwrap();
+
+                    let cell_idx = self.stack.current_cell_index() as usize;
+
+                    if !(cell_idx > contents.cell_count()) {
+                        break;
+                    }
+                }
+            }
+
+            let cell_idx = self.stack.current_cell_index() as usize;
+
+            assert!(cell_idx <= contents.cell_count(),);
+            assert!(!contents.is_leaf());
+
+            if cell_idx == contents.cell_count() {
+                // Move to right child
+                // should be safe as contents is not a leaf page
+                let right_most_pointer = contents.rightmost_pointer().unwrap();
+                self.stack.advance();
+                let mem_page = self.pager.read_page(right_most_pointer as usize)?;
+                self.going_upwards = false;
+                self.stack.push(mem_page);
+            } else {
+                // Move to child left page
+                let cell = contents.cell_get(
+                    cell_idx,
+                    payload_overflow_threshold_max(
+                        contents.page_type(),
+                        self.usable_space() as u16,
+                    ),
+                    payload_overflow_threshold_min(
+                        contents.page_type(),
+                        self.usable_space() as u16,
+                    ),
+                    self.usable_space(),
+                )?;
+
+                match cell {
+                    BTreeCell::TableInteriorCell(TableInteriorCell {
+                        _left_child_page: left_child_page,
+                        ..
+                    })
+                    | BTreeCell::IndexInteriorCell(IndexInteriorCell {
```

Comment:
since we are only handling int-key tables, this branch is completely redundant right now and we can just use

```
let left_child_page = contents.cell_table_interior_read_left_child_page(cell_idx)
```